File: coding-for-fun/elixir/celsius_to_fahrenheit.exs
# author: https://github.com/AnotherGithubDude, 2022
# licence: cc0
# tested with Elixir 1.14.2 (compiled with Erlang/OTP 25)

# Prompt user to input the Celsius value from command-line
# Formula example: (23,333°C × 9/5) + 32 = 74°F
f_input = String.trim(IO.gets("Please enter the temperatur in Celcius (30.0, 23.3):\t"))

# Convert the Fahrenheit input to Celsius and save the result
pre_result = String.to_float(f_input)
result = (pre_result * 9/5) + 32

# Print the result as output on the terminal 
IO.puts("\nThe temperature in Fahrenheit is: #{result}")

File: coding-for-fun/go/celsius_to_fahrenheit.go
package main
import "fmt"
// author: https://github.com/AnotherGithubDude, 2022
// tested with: go1.19.4

func main() {
// Prompt user to input the Celsius value from command-line
fmt.Println("Please enter the temperature in Celsius (23.5, 30.0):\t")
var c_input float64
_, err := fmt.Scanf("%f", &c_input)
if err != nil {
	fmt.Println(err)
}

// Convert the Celsius input to Fahrenheit and save the result
pre_result := c_input
result := (pre_result * 1.8) + 32

// Print the result as output on the terminal 
fmt.Println("The temperature in Fahrenheit is: ", result)
}
